### DISCUSSION TITLE
passability/access model coding and descriptions

### DISCUSSION THREAD

**Discussion by smnorris** [smnorris/bcfishpass#221] [Ideas]
RE https://github.com/smnorris/bcfishpass/issues/181

Replacing/clarifying 'POTENTIALLY ACCESSIBLE' etc codes is really two separate questions:
- how best to code passability/access in the `streams`/`crossings` tables
- define/standardize our language around passability/access model results

My latest comments in the issue thread:

## passability/access codes in data
From my perspective, the cleanest would be something like this:

`observations_upstr_<spp>` - list of observations upstream for given species scenario
`barriers_natural_dnstr_<spp>` - list of known/modelled downstream natural barriers for given species scenario
`barriers_anthropogenic_known_dnstr` - list of known downstream anthropogenic barriers (PSCIS assessed barriers and dams) 
`barriers_anthropogenic_modelled_dnstr` - list of downstream modelled/potential anthropogenic barriers (road/stream crossings)
`barriers_remediated_dnstr` - list of remediations immediately downstream (does not affect the passability status, this is more for reporting)

## passability/access descriptions
From the codes, the current status of the stream can be determined. Having the status spelled out in the data may or may not be useful, depends on the user.  For display purposes I might get even more specific:

- no known natural barrier downstream
- no known natural barrier downstream, modelled potential barrier downstream
- no known natural barrier downstream, known anthropogenic barrier downstream

Note that with the above coding scenario we would be lumping all dams into the `barriers_anthropogenic_known_dnstr` - special handling of large hydro dams would be done at the reporting level rather than within the model. I don't see any reason to differentiate them if all dams (with no known fish passage measures) are considered to be barriers. Historic habitat is interesting to visualize (and our current `bcdams` dataset is not good enough to make this distinction reliably anyway, hopefully CABD updates will improve this)

**Comment by NewGraphEnvironment** [comment]
oh jeez. Bit of a head spinner. I think 

https://github.com/smnorris/bcfishpass/issues/181#issuecomment-1080858949 

makes sense. We may as well have a very explicit column that allows the user to quickly get a grip on the big picture status of a segment all at once.  As far as the users ability to interpret the pdf mapping though, we obviously might want to keep the mapped streamline symbology simple and potentially as it is?

**Comment by smnorris** [comment]
Very happy to keep the symbology as is, I don't want to change that.
But I think the legend could read "No known natural barrier downstream" vs "Accessible".
Replacing "Potentially accessible" with "No known natural barrier downstream, modelled potential barrier downstream" takes up a lot of space in a legend but may be worth stuffing it in?

For the single summary column, @captainmarmot and I have tentatively scheduled next Monday afternoon to discuss further. Maybe you can join in?

**Comment by smnorris** [comment]
tagging @nickw-CWF, in case you aren't already seeing this.
